### ISSUE TITLE
Settings Validated Red X in 2.1 RC1 and Battery Voltage Displays No Decimals

### ISSUE THREAD

**Issue by Nihilistic** [iNavFlight/inav#4207]
## Current Behavior

After setting up the OSD on both my F405-CTR and F722-SE, and plugging in a flight battery for the first test flight, I got a navigation is unsafe voltage something or other error flashing on my OSD, and only the whole numbers of my battery voltage would display. Workaround is selecting 2 decimal places from the dialog box on the OSD tab, clicking save, and then selecting back to 1 decimal place, and clicking save, and then voltage displays correctly on OSD.

Possibly related to #3507
## Steps to Reproduce

1.
2.
3.
4.

## Additional context

---

- FC Board name and vendor:
- INAV version string:

**Comment by giacomo892** [comment]
If settings are no validated you likely have not upgraded following the guide on the release notes.

**Comment by Nihilistic** [comment]
I always do a full chip erase and do the full configuration manually. No CLI dump pastings at all.

After clicking on the dialog box and bumping decimals up to 2 and clicking save, and then selecting 1 decimal again and clicking save, the settings validated issue goes away. Only have to do this once, not every time I plug in the battery.

It's a minor thing, but thought I'd mention it.

**Comment by DzikuVx** [comment]
![image](https://user-images.githubusercontent.com/966811/51210494-396c9080-1913-11e9-9fff-81d4a5c4cfba.png)

![image](https://user-images.githubusercontent.com/966811/51210612-8fd9cf00-1913-11e9-8f96-af8df41b896b.png)

Just tested on F405-CTR and it is working correctly with Full Chip Erase, settings are valid after flashing

**Comment by Nihilistic** [comment]
LOL...oh I must have imagined it then. 

You didn't go through the whole configuration and set up the OSD. Didn't even apply the quadx mixer. 

I'm using PAL, don't know if that makes a difference. And the Split 2S on this bird.

**Comment by fiam** [comment]
@Nihilistic can you post a config dump that, when loaded, reproduces the problem? I've checked the code and it seems to me that setting is correctly initialised on first boot. See https://github.com/iNavFlight/inav/blob/development/src/main/io/osd.c#L2566

**Comment by Nihilistic** [comment]
I'm using the modified hex that fixes the Split 2S control issue that was just posted today. But, I've had this problem before using the official 2.1 RC1 hex. Maybe it's something very specific to how I'm setting up my OSD (which fields I've chosen and where I've placed them). Here's my most recent dump, but it's after I did the workaround, so I don't know if it will show anything.
[Fully Configured Dump.txt](https://github.com/iNavFlight/inav/files/2761573/Fully.Configured.Dump.txt)
